legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/AppendOnlyPlanner.java

```diff
@@ -90,6 +111,11 @@ public LogicalPlan buildLogicalPlanForIngest(Resources resources)
         List<Value> fieldsToSelect = new ArrayList<>(dataFields);
         List<Value> fieldsToInsert = new ArrayList<>(dataFields);
 
+        // Add digest generation (if applicable)
+        List<Value> stringFieldValues = fieldsToSelect.stream().map(value -> FunctionImpl.builder().functionName(FunctionName.CONVERT).addValue(value, ObjectValue.of(DataType.VARCHAR.name())).build()).collect(Collectors.toList());
```

Comment:
Please double check if stringFieldValues is really needed here
This seems very specific to H2, let's move CONVERT logic to H2 Visitor.

Reply:
Sure, changed!